### DISCUSSION TITLE
2 questions

### DISCUSSION THREAD

**Discussion by techcatgato** [pybricks/support#1773] [Q&A]
1. Is there any way to make a "background task" that will always run  in the background?
2. How does pybricks calculate distance for drive bases? Is it reliable? Will it be affected by turns like pivot turns or PID?

**Comment by laurensvalk** [comment]
> Is there any way to make a "background task" that will always run in the background?

Yes, you can use the multitask block or `multitask` Python command.

>How does pybricks calculate distance for drive bases?

It is the average angle between the left and right motor. Then it uses the provided wheel diameter to scale it to a distance. So it is only as accurate as the rotation sensors. The wheels may slip and compress, so it is only an estimate.
